sys/auto_init/saul/auto_init_sht1x_saul.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization for SHT1X temperature/humidity sensors
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#ifdef MODULE_SHT1X
+
+#include "log.h"
+#include "saul_reg.h"
+#include "sht1x_params.h"
+#include "sht1x.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define SHT1X_NUM     (sizeof(sht1x_params) / sizeof(sht1x_params[0]))
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+extern sht1x_dev_t sht1x_devs[SHT1X_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[SHT1X_NUM * 2];
+
+/**
+ * @name    Import SAUL endpoints
+ * @{
+ */
+extern const saul_driver_t sht1x_saul_temp_driver;
+extern const saul_driver_t sht1x_saul_hum_driver;
+/** @} */
+
+void auto_init_sht1x_saul(void)
```

Comment:
No need for a saul specific function, you could add it to `auto_init_sht1x` but guard it with an `#ifdef` if the saul module is loaded. If you do this, `auto_init_sht1x_saul.c` can be removed. And the `auto_init_sht1x.c` could be with the other saul devices auto init files.